feature-libs/asm/root/interceptors/user-id.interceptor.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Inject, Injectable } from '@angular/core';
+import {
+  OCC_ASM_TOKEN,
+  OCC_USER_ID_CONSTANTS,
+  UserIdService,
+} from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { concatMap, map } from 'rxjs/operators';
+
+/**
+ * Looks for a specific key in the HttpRequest's context (OCC_ASM_TOKEN) to decide when to
+ * configure a request with 'sap-commerce-cloud-user-id' header.
+ */
+@Injectable({ providedIn: 'root' })
+export class UserIdInterceptor implements HttpInterceptor {
+  private readonly userIdHeader = 'sap-commerce-cloud-user-id';
+
+  private readonly uniqueUserIdConstants: Set<string>;
+
+  constructor(
+    protected userIdService: UserIdService,
+    @Inject(OCC_USER_ID_CONSTANTS)
+    userIdConstants: { [identifier: string]: string }
+  ) {
+    this.uniqueUserIdConstants = new Set(Object.values(userIdConstants));
+  }
+
+  intercept(
+    httpRequest: HttpRequest<unknown>,
+    next: HttpHandler
+  ): Observable<HttpEvent<unknown>> {
+    const asmContext = httpRequest.context.get(OCC_ASM_TOKEN);
+
+    let userIdObservable: Observable<string | undefined>;
+
+    if (typeof asmContext.sendUserIdAsHeader === 'string') {
+      userIdObservable = of(asmContext.sendUserIdAsHeader);
+    } else if (asmContext.sendUserIdAsHeader) {
+      userIdObservable = this.userIdService
+        .getUserId()
```

Comment:
consider calling `takeUserId()` instead of `getUserId()` if we just need the current userId and move on.

Reply:
! I didn't realize the difference. Thank you.